spec/models/ability_spec.rb

```diff
@@ -1,437 +1,240 @@
+        it { expect(subject).not_to be_able_to [:update, :destroy], other_team}
       end
 
-
-      describe 'who is allowed to see email address in user profile' do
-
-        # email address is hidden: admin, user's supervisor in current season (confirmed)
-        # email address is not hidden: admin, confirmed user, user him/herself
-
-        let(:other_user) { create(:user) }
-
-        context 'when email address is hidden' do
-          context 'when an admin' do
-            before do
-              allow(user).to receive(:admin?).and_return(true)
-              allow(ability).to receive(:supervises?).with(other_user, user).and_return(false)
-            end
-            it 'allows to see hidden email address' do
-              other_user.hide_email = true
-              expect(ability).to be_able_to(:read_email, other_user)
-            end
-          end
-
-          context "when user's supervisor in current season (confirmed)" do
-            before do
-              allow(user).to receive(:admin?).and_return(false)
-              allow(ability).to receive(:supervises?).with(other_user, user).and_return(true)
-              allow(user).to receive(:confirmed?).and_return(true)
-            end
-            it 'allows to see hidden email address' do
-              other_user.hide_email = true
-              expect(ability).to be_able_to(:read_email, other_user)
-            end
-          end
-        end
-
-        context 'when email address is not hidden' do
-          context 'when an admin' do
-            before do
-              allow(user).to receive(:admin?).and_return(true)
-              allow(user).to receive(:confirmed?).and_return(false)
-            end
-            it 'allows to see not hidden email address' do
-              other_user.hide_email = false
-              expect(ability).to be_able_to(:read_email, other_user)
-            end
-          end
-
-          context 'when a confirmed user' do
-            before do
-              allow(user).to receive(:admin?).and_return(false)
-              allow(user).to receive(:confirmed?).and_return(true)
-            end
-            it 'allows to see not hidden email address' do
-              other_user.hide_email = false
-              expect(ability).to be_able_to(:read_email, other_user)
-            end
-          end
-
-          context 'when the user him/herself' do
-            it 'allows to see not hidden email address' do
-              user.hide_email = false
-              expect(ability).to be_able_to(:read_email, user)
-            end
-          end
+      describe "Student" do
+        let(:student_team) { build(:team, :in_current_season) }
+        let(:second_team) { build(:team, :in_current_season) }
+        before do
+          allow(user).to receive(:current_student?).and_return(true)
         end
+        # TODO Question: is this ^ stub correct to test the ability only?
```

Comment:
Given that you know what `current_student?` actually does and that it's tested elsewhere: then yes ✅